custom/icds_reports/ucr/data_sources/ccs_record_cases.json

```diff
@@ -503,6 +503,29 @@
           }
         }
       },
+      {
+        "column_id": "person_case_id",
+        "datatype": "string",
+        "display_name": null,
+        "expression": {
+          "name": "parent_id",
+          "type": "named"
+        },
+        "is_nullable": true,
+        "is_primary_key": false,
+        "transform": {},
+        "type": "expression"
+      },
+      {
+        "display_name": null,
+        "property_path": [],
+        "datatype": "date",
+        "is_primary_key": false,
+        "column_id": "pnc1_date",
+        "is_nullable": true,
+        "type": "raw",
+        "property_name": "pnc1_date"
```

Comment:
from where I can fill this property. I can fill the `person_case_id` from the monthyl table but from where to fill `pnc1_date`

Reply:
This is a bit hard as in ccs_record_monthly, we only store some derivative data. I think that we can work around this though, there are two options.

1. If we only care about `pnc1_date` when `pnc_complete = 1` in ccs_record_monthly, we could rebuild only the records that have `pnc_complete = 1`. basic pseudo code would look like:

```python
for ccs_record in ccs_record_monthly.where(pnc_complete = 1):
    AsyncIndicator.queue(ccs_record)
```

2. From the [existing UCR](https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/ucr/data_sources/ccs_record_cases_monthly_tableau2.json#L1733-L1759) we only care about pnc_complete which is true if `pnc1_date < month`, so we could do something that fakes ok data for now. For example:

```sql
UPDATE ccs_record_cases_ucr  ccs_ucr SET
   pnc1_date = '2019-01-01'
FROM ccs_record_monthly_tableau AS monthly
WHERE monthly.pnc_complete = 1 AND monthly.month = '2019-01-01'  AND ccs_ucr.doc_id = monthly.doc_id AND ccs_ucr.pnc1_date IS NULL
```

This won't put the exactly correct information in pnc1_date, but for the purposes of the aggregation it should be fine as we only care if pnc1_date was before a particular month instead of the actual date for now.

The tradeoff here is that #1 would be faster for dev time and be more correct if pnc1_date is used later. #2 would be faster to run and allow the other PR to be merged sooner (I think)

Reply:
1st one seems reasonable to me because in future when the actual date will be required. we might not remember that the dates are fakes and it will be worse if someone else picks that future work other than us. he/She wont have any context of fake dates. 
Restarted the build. Once passes. Good  for final review. Will run the rebuild UCR code after it merges.

one question, above pseudo code you have written for 1. will it put the pnc_date in to the ucr table?
@emord

Reply:
yes. option 1 works because it puts each document in our celery build queue (AsyncIndicator) and it will recalculate the whole document as it can. Could you update this PR with the management command to do that? (since what I put above is only psuedo code)